docs/why-a-workspace.md

```diff
@@ -0,0 +1,69 @@
+- When a developer goes to work on Application A they need to be aware of 
+which version the other applications are designed to work with
+- ...and they need to have those in their local environment
+- ...and many others (multiple instances of node_modules, complex scripts to launch all apps, etc)...
+
+A workspace (mono repo) for the application domain can help alleviate a lot of those
+challenges and it can help facilitate positive programming and architecture practices.
+
+Here's an example. Say we have some code to format support ticket metadata that we use
+in the customer app and the admin app. Making a shared lib folder to house that code in
+our workspace is trivial. We can move the code there and `import` reference it within
+our apps. No need to `npm install` the library for each app. And when we decide to extract
+out some common string formatting code that is within that library...again, a trivial new lib
+folder creation and code move and we are on our way.
+
+Because that library creation and refactoring is trivial, we are not hesitant to take it
+on. So we can find ourselves more apt to modularize our code as we architect it because
+the barrier to entry is minimal. And the more we modularize the more we can take advantage
+of things like lazy loading and tree shaking and more. Things that are important in
+the web space.
+
+## Advantages
+Below are some more specific advantages to going with a mono repo workspace.
+
+- Unified versioning
+    - Everything at that current commit works together
+    - A label or branch can capture the same
+- Promotes code sharing and reuse
+    - Easy to split code into lib modules
+    - Easy to consume/implement that code and the latest changes to it
+- Easier dependency management
+    - One node_modules for all code
+    - One build setup (like the AngularCLI)
+- Refactoring benefits
+    - Code editors and IDEs are "workspace" aware
+    - Can have a single commit for a refactor that spans applications in the domain
+- Consistent developer experience
+    - Ensures all necessary dependant code is available
+
+## Disadvantages
+Now look, we are talking real world application development here. And as we know, there 
+are no 100% perfect fits when it comes to development. We have to weight the pluses and minuses
+and make decisions accordingly. So, here are some of the potential disadvantages that
+we want to be aware of when it comes to a mono repo workspace.
+
+- Takes work to try and limit access to parts of the code base
+- An upgrade to a lib requires a change to all implementors (can't roll out different versions side by side)
+- Can lead to accepted dependencies, making it overkill to work on a small feature
+    - Say some library code is designed to hit the web service, in a single workspace
+      you know the service is available so effort might not be made to be able to run
+      that library code with a mock service
+- The need for build optimization and other tooling
```

Comment:
I think this need is present regardless of the monorepo. 

monorepo is a way to organize code, it doesn't affect how the code gets bundled and shipped

Reply:
Removed that bullet point